src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -316,16 +334,21 @@ private void start() {
       Queryable db = new MetricsDBProvider();
       ThresholdMain thresholdMain = new ThresholdMain(RcaConsts.THRESHOLDS_PATH, rcaConf);
       Persistable persistable = PersistenceFactory.create(rcaConf);
+      networkThreadPoolReference.set(buildNetworkThreadPool(rcaConf.getNetworkQueueLength()));
```

Comment:
It will still need a restart of some sort. The queue length is only read when starting/restarting the controller. Essentially, this lets us change queue size without needing a JAR replacement, just modifying the RCA conf file and restarting should suffice.

> What happens when the old executor was executing and this thread sees a configuration change and tries to update the pool, will we ever be running the same code across two threads ?

when we restart, we stop the old threadpool executor. This scenario should not happen.